server-spi/src/main/java/org/keycloak/models/KeycloakContext.java

```diff
@@ -55,6 +55,12 @@ public interface KeycloakContext {
 
     HttpHeaders getRequestHeaders();
 
+    /**
+     * Will always return null
+     *
+     * @deprecated
+     */
+    @Deprecated
```

Comment:
```suggestion
    /**
     * Will always return null
     *
     * @deprecated obtaining contextual data is limited and you should refer to a current context provider (f.e. Resteasy.getContextData)
     */
    @Deprecated(forRemoval = true)
```

We should possibly explain more current status, right? I find marking `forRemoval=true` more visible in the deprecation process. @shawkins WDYT?

Reply:
Ok, updated with more text. In this method we don't even want to refer them to Resteasy.getContextData as that is only for obtaining the KeycloakSession, which they should already have for this KeycloakContext.

Reply:
Nice. Yep, I know, the intention behind that was to point users somewhere for more info, but it is better in this form.